Enable stacked scales and with reverse option support (#655)
chartjs/chartjs-plugin-annotation: docs/guide/migrationV2.md

 * `xScaleID` option default has been changed, now set to `undefined`. If the option is missing, the plugin will try to use the first scale of the chart, configured as `'x'` axis. If more than one scale has been defined in the chart as `'x'` axis, the option is mandatory to select the right scale.
 * `yScaleID` option default has been changed, now set to `undefined`. If the option is missing, the plugin will try to use the first scale of the chart, configured as `'y'` axis. If more than one scale has been defined in the chart as `'y'` axis, the option is mandatory to select the right scale.

>  * The scale dimension, not chart area anymore, is used to set `xMin`, `xMax`, `yMin` and `yMax` options when missing. This enables the stacked scale management.

Reviewer: Should add a link to the chart.js docs for "stacked scale", because it might be confused with normal stacking.
Also could be worth mentioning that this only affects charts with stacked scales.

Author: have a look if sounds better now

 * `xScaleID` option default has been changed, now set to `undefined`. If the option is missing, the plugin will try to use the first scale of the chart, configured as `'x'` axis. If more than one scale has been defined in the chart as `'x'` axis, the option is mandatory to select the right scale.
 * `yScaleID` option default has been changed, now set to `undefined`. If the option is missing, the plugin will try to use the first scale of the chart, configured as `'y'` axis. If more than one scale has been defined in the chart as `'y'` axis, the option is mandatory to select the right scale.

>  * When [stacked scales](https://www.chartjs.org/docs/latest/axes/cartesian/#common-options-to-all-cartesian-axes) are used and  `xMin`, `xMax`, `yMin` and `yMax` options are missing, the scale dimension is used instead of chart area to set the annotation dimension.
>  * When [stacked scales](https://www.chartjs.org/docs/latest/axes/cartesian/#common-options-to-all-cartesian-axes) are used and  `xValue` and `yValue` options are missing, the scale dimension is used instead of chart area to set the annotation location.

Reviewer: ```suggestion
* When [stacked scales](https://www.chartjs.org/docs/latest/axes/cartesian/#common-options-to-all-cartesian-axes) are used, instead of the whole chart area, the designated scale area is used as fallback for `xMin`, `xMax`, `yMin`, `yMax`, `xValue` or `yValue` options.
```

Not sure if this is any better. @etimberg what do you think?

Reviewer 2: That makes sense to me
